repair/row_level.cc

```diff
@@ -2429,6 +2435,52 @@ future<> repair_service::init_ms_handlers() {
     ms.register_repair_get_diff_algorithms([] (const rpc::client_info& cinfo) {
         return make_ready_future<std::vector<row_level_diff_detect_algorithm>>(suportted_diff_detect_algorithms());
     });
+    ms.register_repair_update_system_table([this] (const rpc::client_info& cinfo, repair_update_system_table_request req) -> future<repair_update_system_table_response> {
+        auto from = cinfo.retrieve_auxiliary<gms::inet_address>("baddr");
+        rlogger.debug("repair[{}]: Got repair_update_system_table_request from node={}, range={}, repair_time={}", req.repair_uuid, from, req.range, req.repair_time);
+        auto& db = this->get_db();
+        co_await db.invoke_on_all([&req] (database& local_db) {
+            auto& table = local_db.find_column_family(req.table_uuid);
+            return ::update_repair_time(table.schema(), req.range, req.repair_time);
+        });
+        sstring cql = format("INSERT INTO system.{} (table_uuid, repair_time, repair_uuid, keyspace_name, table_name, range_start, range_end) VALUES (?, ?, ?, ?, ?, ?, ?)",
+                db::system_keyspace::REPAIR_HISTORY);
+        auto range_start = req.range.start() ? req.range.start()->value() : dht::minimum_token();
+        auto range_end = req.range.end() ? req.range.end()->value() : dht::maximum_token();
+        db_clock::time_point ts = db_clock::from_time_t(gc_clock::to_time_t(req.repair_time));
+        co_await db::qctx->execute_cql(cql, req.table_uuid, ts, req.repair_uuid, req.keyspace_name, req.table_name,
+                dht::token::to_int64(range_start), dht::token::to_int64(range_end)).discard_result();
+        co_return repair_update_system_table_response();
+    });
+    ms.register_repair_flush_hints_batchlog([this] (const rpc::client_info& cinfo, repair_flush_hints_batchlog_request req) -> future<repair_flush_hints_batchlog_response> {
+        auto from = cinfo.retrieve_auxiliary<gms::inet_address>("baddr");
+        rlogger.info("repair[{}]: Started to process repair_flush_hints_batchlog_request from node={}, target_nodes={}, hints_timeout={}s, batchlog_timeout={}s",
+                req.repair_uuid, from, req.target_nodes, req.hints_timeout.count(), req.batchlog_timeout.count());
+        std::vector<gms::inet_address> target_nodes(req.target_nodes.begin(), req.target_nodes.end());
+        db::hints::sync_point sync_point = co_await _sp.local().create_hint_sync_point(std::move(target_nodes));
```

Comment:
What about flushing MV hints?

Reply:
@piodul Does the hint sync point api flush mv hints?

Reply:
Yes - what you are doing now will make sure that both regular and MV hints are flushed.

Reply:
Cool. Thanks!